### PR TITLE
samples: matter: Used DFUsync module in the DFU over SMP

### PR DESCRIPTION
**Pull request by kkasperczyk-no**
Updated Matter repository to pull the change introducing DFUSync module and use in the DFU over SMP implementation in Matter.

### CODE REVIEW

File: samples/matter/common/src/dfu/smp/dfu_over_smp.cpp [hunk 1]
```diff
@@ -30,29 +32,36 @@ constexpr uint16_t kAdvertisingIntervalMin = 400;
...
+static uint32_t sDfuSyncMutexId;
+
 namespace
 {
-enum mgmt_cb_return UploadConfirmHandler(uint32_t event,
-					 enum mgmt_cb_return prev_status,
-					 int32_t *rc, uint16_t *group,
-					 bool *abort_more, void *data,
-					 size_t data_size)
+enum mgmt_cb_return UploadConfirmHandler(uint32_t event, enum mgmt_cb_return prev_status, int32_t *rc, uint16_t *group,
```
Comment (ArekBalysNordic) [inline samples/matter/common/src/dfu/smp/dfu_over_smp.cpp:40]: @markaj-nordic  it's yet another place where we can have different clang-format configs.

**Review by ArekBalysNordic — APPROVED** [review]
This part looks good.

File: samples/matter/common/src/dfu/smp/dfu_over_smp.cpp (hunk 1 above)
Comment (kkasperczyk-no) [reply samples/matter/common/src/dfu/smp/dfu_over_smp.cpp:40]: So is it correct now, or my configuration is wrong?

Comment (markaj-nordic) [reply samples/matter/common/src/dfu/smp/dfu_over_smp.cpp:40]: Not sure, maybe the formatting here was invalid in the first place.

File: samples/matter/common/src/dfu/smp/dfu_over_smp.cpp [hunk 2]
```diff
@@ -63,6 +72,14 @@ enum mgmt_cb_return CommandHandler(uint32_t event,
 	return MGMT_CB_OK;
 }
 
+enum mgmt_cb_return DfuStoppedHandler(uint32_t event, enum mgmt_cb_return prev_status, int32_t *rc, uint16_t *group,
```
Comment (markaj-nordic) [inline samples/matter/common/src/dfu/smp/dfu_over_smp.cpp:75]: Although our compiler does not complain about it, maybe it's still better to tag all arguments are unused or skip their names.

File: samples/matter/common/src/dfu/smp/dfu_over_smp.cpp [hunk 3]
```diff
@@ -130,4 +158,24 @@ void DFUOverSMP::StartServer()
...
 
+void DFUOverSMP::Disconnected(bt_conn *conn, uint8_t reason)
+{
+	if (!sDfuInProgress) {
+		return;
+	}
+
+	bt_conn_info btInfo;
+
+	bt_conn_get_info(conn, &btInfo);
```
Comment (markaj-nordic) [inline samples/matter/common/src/dfu/smp/dfu_over_smp.cpp:169]: This function can return an error on failure, it's worth to consider such a case.

**Review by markaj-nordic — COMMENTED** [review]
@kkasperczyk-no do you think it requires the release notes update?

File: samples/matter/common/src/dfu/smp/dfu_over_smp.cpp (hunk 2 above)
Comment (kkasperczyk-no) [reply samples/matter/common/src/dfu/smp/dfu_over_smp.cpp]: Ok, I remove the unused argument names, also for the already existing callbacks.

File: samples/matter/common/src/dfu/smp/dfu_over_smp.cpp (hunk 3 above)
Comment (kkasperczyk-no) [reply samples/matter/common/src/dfu/smp/dfu_over_smp.cpp]: Done

**Comment by kkasperczyk-no** [comment]
> @kkasperczyk-no do you think it requires the release notes update?

TBH I don't think so, how about you? It's not a feature, rather a fix for bug existing from the very beginning, but we also didn't have known issue for that.

**Comment by markaj-nordic** [comment]
> > @kkasperczyk-no do you think it requires the release notes update?
> 
> TBH I don't think so, how about you? It's not a feature, rather a fix for bug existing from the very beginning, but we also didn't have known issue for this bug.

Yehh, that's fair. It'd make sense if someone used this "feature" of doing 2 DFUs in parallel which I don't think is much probable :)

**Review by markaj-nordic — APPROVED** [review]

**Pull request merged**